src/org/zaproxy/zap/extension/fuzz/payloads/generator/RegexPayloadGenerator.java

```diff
@@ -209,12 +222,16 @@ public boolean hasNext() {
             if (maxPayloads > 0 && count >= maxPayloads) {
                 return false;
             }
-            return iterator.hasNext();
+            return randomOrder ? true : iterator.hasNext();
```

Comment:
Any reason to not rely on the `Iterator`? It would not require adding ifs here and in `next`, which are called more often than `reset`.

Reply:
I'm not following, using the `Iterator` for the random part does not change the current logic.

Reply:
yeah, you are right :)